contrib/golang/filters/http/source/cgo.cc

```diff
@@ -0,0 +1,134 @@
+#include "contrib/golang/filters/http/source/golang_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Golang {
+
+//
+// These functions may be invoked in another go thread,
+// which means may introduce race between go thread and envoy thread.
+//
+
+absl::string_view copyGoString(void* str) {
+  if (str == nullptr) {
+    return "";
+  }
+  auto goStr = reinterpret_cast<GoString*>(str);
+  return absl::string_view(goStr->p, goStr->n); // NOLINT(modernize-return-braced-init-list)
+}
+
+extern "C" {
+
+void moeHandlerWrapper(void* r, std::function<void(std::shared_ptr<Filter>&)> f) {
+  auto req = reinterpret_cast<httpRequestInternal*>(r);
+  auto weakFilter = req->weakFilter();
+  if (auto filter = weakFilter.lock()) {
+    f(filter);
+  }
+}
+
+void moeHttpContinue(void* r, int status) {
+  moeHandlerWrapper(r, [status](std::shared_ptr<Filter>& filter) {
+    filter->continueStatus(static_cast<GolangStatus>(status));
+  });
+}
+
+void moeHttpSendLocalReply(void* r, int response_code, void* body_text, void* headers,
+                           long long int grpc_status, void* details) {
+  moeHandlerWrapper(r, [response_code, body_text, headers, grpc_status,
+                        details](std::shared_ptr<Filter>& filter) {
+    (void)headers;
```

Comment:
nit: use UNREFERENCED_PARAMETER or just delete the parameter name from the call signature. Here and elsewhere.

Reply:
Done.